sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/implementation/SharesImpl.java

```diff
@@ -94,7 +94,7 @@ public SharesImpl(AzureFileStorageImpl client) {
         @Put("{shareName}")
         @ExpectedResponses({201})
         @UnexpectedResponseExceptionType(StorageErrorException.class)
-        Mono<SharesCreatePermissionResponse> createPermission(@PathParam("shareName") String shareName, @HostParam("url") String url, @QueryParam("timeout") Integer timeout, @HeaderParam("x-ms-version") String version, @QueryParam("restype") String restype, @QueryParam("comp") String comp, Context context);
+        Mono<SharesCreatePermissionResponse> createPermission(@PathParam("shareName") String shareName, @HostParam("url") String url, @QueryParam("timeout") Integer timeout, @HeaderParam("x-ms-version") String version, @BodyParam("application/json; charset=utf-8") SharePermission sharePermission, @QueryParam("restype") String restype, @QueryParam("comp") String comp, Context context);
```

Comment:
This is how to tell RestProxy to send JSON. Response deserialization is smarter- it will use JSON or XML parser by peeking into the response body.